File: WayFinder/Models/OutfitWeatherModels.swift
import Foundation

// MARK: - Outfit Model
struct Outfit: Decodable, Identifiable {
    let id: String
    let userId: String
    let bookingId: String
    let imageUrl: String
    let detectedItems: [String]
    let weatherData: WeatherData?
    let recommendation: OutfitRecommendation?
    let isApproved: Bool
    let createdAt: String?
    let updatedAt: String?
    let outfitDate: String?
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case userId = "user_id"
        case bookingId = "booking_id"
        case imageUrl = "image_url"
        case detectedItems = "detected_items"
        case weatherData = "weather_data"
        case recommendation
        case isApproved = "is_approved"
        case createdAt
        case updatedAt
        case outfitDate = "outfit_date"
    }
}

// MARK: - Weather Data Model
struct WeatherData: Decodable {
    let temperature: Int
    let condition: String
    let humidity: Int?
    let windSpeed: Double?
    
    enum CodingKeys: String, CodingKey {
        case temperature
        case condition
        case humidity
        case windSpeed = "wind_speed"
    }
}

// MARK: - Outfit Recommendation Model
struct OutfitRecommendation: Decodable {
    let isSuitable: Bool
    let score: Int
    let feedback: String
    let suggestions: [String]
    
    enum CodingKeys: String, CodingKey {
        case isSuitable = "is_suitable"
        case score
        case feedback
        case suggestions
    }
}

// MARK: - Analyze Outfit Request
struct AnalyzeOutfitRequest: Encodable {
    let bookingId: String
    let imageUrl: String
    
    enum CodingKeys: String, CodingKey {
        case bookingId = "booking_id"
        case imageUrl = "image_url"
    }
}

// MARK: - Upload Outfit Response
struct UploadOutfitResponse: Decodable {
    let message: String
    let imageUrl: String
    let analysis: Outfit
    
    enum CodingKeys: String, CodingKey {
        case message
        case imageUrl = "image_url"
        case analysis
    }
}

File: app/src/main/java/tn/esprit/wayfinder/models/OutfitWeatherModels.kt
package tn.esprit.wayfinder.models

import kotlinx.serialization.Serializable

@Serializable
data class Outfit(
    val _id: String? = null,
    val user_id: String,
    val booking_id: String,
    val image_url: String,
    val detected_items: List<String> = emptyList(),
    val weather_data: WeatherData? = null,
    val recommendation: OutfitRecommendation? = null,
    val is_approved: Boolean = false,
    val createdAt: String? = null,
    val updatedAt: String? = null
)

@Serializable
data class WeatherData(
    val temperature: Int,
    val condition: String,
    val humidity: Int? = null,
    val wind_speed: Double? = null
)

@Serializable
data class OutfitRecommendation(
    val is_suitable: Boolean,
    val score: Int,
    val feedback: String,
    val suggestions: List<String> = emptyList()
)

@Serializable
data class AnalyzeOutfitRequest(
    val booking_id: String,
    val image_url: String
)

@Serializable
data class UploadOutfitResponse(
    val message: String,
    val image_url: String,
    val analysis: Outfit
)

